Autonomous cannot be selected without code in robotcontainer/dashboard.

A sendable chooser needs to be implemented or a replacement needs to be installed.

```
make sim
python robot.py sim
20:43:32:395 INFO    : wpilib              : WPILib version 2022.2.1.5
20:43:32:395 INFO    : wpilib              : HAL version 2022.2.1.1
20:43:32:395 INFO    : wpilib              : Running with simulated HAL.
20:43:32:396 INFO    : wpilib              : robotpy-wpimath version 2022.2.1.4
20:43:32:396 INFO    : wpilib              : pyntcore version 2022.2.1.2
20:43:32:396 INFO    : wpilib              : robotpy-commands-v2 version 2022.2.1.1
20:43:32:396 INFO    : wpilib              : robotpy-wpiutil version 2022.2.1.1
20:43:32:396 INFO    : wpilib              : robotpy-halsim-gui version 2022.2.1.0
20:43:32:396 INFO    : wpilib              : robotpy-rev version 2022.0.1
20:43:32:396 INFO    : wpilib              : robotpy-rev-color version 2021.1.2.0
20:43:32:396 INFO    : faulthandler        : registered SIGUSR2 for PID 5528
20:43:32:396 INFO    : halsim_gui          : WPILib HAL Simulation 2022.2.1.0
HAL Extensions: Attempting to load: libhalsim_gui
Simulator GUI Initializing.
Simulator GUI Initialized!
HAL Extensions: Successfully loaded extension
20:43:32:542 WARNING : pyfrc.physics       : Cannot enable physics support, /home/joe/git/pre2022season/rio/physics.py not found
Not loading CameraServerShared

********** Robot program startup complete **********
Default SimulationPeriodic() method... Override me!
20:43:32:778 INFO    : nt                  : server: client CONNECTED: 127.0.0.1 port 39604
20:43:34:485 ERROR   : your.robot          : Unhandled exception
Traceback (most recent call last):
  File "/home/joe/.local/share/virtualenvs/rio-t_s-Nm-J/lib/python3.10/site-packages/wpilib/_impl/start.py", line 124, in _start
    self.robot.startCompetition()
  File "/home/joe/git/pre2022season/rio/robot.py", line 38, in autonomousInit
    self.autonomousCommand = self.container.getAutonomousCommand()
  File "/home/joe/git/pre2022season/rio/robotcontainer.py", line 56, in getAutonomousCommand
    return self.chooser.getSelected()
AttributeError: 'RobotContainer' object has no attribute 'chooser'


Locals at innermost frame:

{'self': <robotcontainer.RobotContainer object at 0x7fc738b8a140>}


20:43:34:485 WARNING : robotpy             : The robot program quit unexpectedly. This is usually due to a code error.
The above stacktrace can help determine where the error occurred.

Warning at Main: Warning: Watchdog not fed within 0.020000s
        at frc::Watchdog::Impl::Main() + 0x22a [0x7fc73d05928a]
        at  + 0xd33c4 [0x7fc73f0a93c4]
        at  + 0x9259 [0x7fc73fd8d259]
        at clone + 0x43 [0x7fc73fea35e3]

Error at PrintLoopOverrunMessage: Error: Loop time of 0.020000s overrun
        at frc::IterativeRobotBase::PrintLoopOverrunMessage() + 0x52 [0x7fc73d013342]
        at frc::Watchdog::Impl::Main() + 0x123 [0x7fc73d059183]
        at  + 0xd33c4 [0x7fc73f0a93c4]
        at  + 0x9259 [0x7fc73fd8d259]
        at clone + 0x43 [0x7fc73fea35e3]

make: *** [Makefile:18: sim] Error 1
```

--- a/.github/workflows/black-formatter.yml
+++ b/.github/workflows/black-formatter.yml
@@ -17,3 +17,6 @@ jobs:
         uses: reviewdog/action-suggester@v1
         with:
           tool_name: blackfmt
+      - name: Fail if actions taken
+        if: steps.action_black.outputs.is_formatted == 'true'
+        run: exit 1

--- a/dashboard/www/index.html
+++ b/dashboard/www/index.html
@@ -11,14 +11,14 @@
      pages simultaneously -->
     <script src="/networktables/networktables.js"></script>
 
+
     <!-- Obviously, you will want to copy this file locally in a real 
      dashboard, as the Driver Station won't have internet access -->
     <script src="https://code.jquery.com/jquery-2.2.2.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.14/d3.min.js"></script>
 
-    <!-- uncomment this is you want to use included utility functions that
-     implement common functionality that you might find useful. Requires
-     that d3.js and jQuery are included first -->
-    <!-- <script src="/networktables/utils.js"></script> -->
+    <!-- utility functions to attach to a sendable chooser -->
+    <script src="/networktables/utils.js"></script>
 
     <!-- utility functions to attach to a camera -->
     <script src="/networktables/camera.js"></script>
@@ -41,7 +41,7 @@
                 </td>
                 <td style="vertical-align: top;">Camera 2<br>
                 </td>
-                <td style="vertical-align: top;">Something else<br>
+                <td style="vertical-align: top;">Autonomous Selector<select id="autonomous"></select><br>
                 </td>
             </tr>
             <tr>
@@ -116,6 +116,9 @@
 
             NetworkTables.addGlobalListener(swerveListener, true);
             console.log("Started")
+
+            // attaches the select element to a SendableChooser object
+            attachSelectToSendableChooser("#autonomous", "Autonomous");
 
             // will load a camera from the robot's IP address on port 5800
             loadCameraOnConnect({

--- a/rio/commands/nullauto.py
+++ b/rio/commands/nullauto.py
@@ -1,0 +1,8 @@
+import commands2
+from subsystems import drivetrain
+
+class Auto(commands2.SequentialoCommandGroup):
+
+    def __init__(self):
+        super().__init__()
+        pass

--- a/rio/commands/nullauto.py
+++ b/rio/commands/nullauto.py
@@ -1,8 +1,8 @@
 import commands2
 from subsystems import drivetrain
 
-class Auto(commands2.SequentialoCommandGroup):
 
+class Auto(commands2.SequentialoCommandGroup):
     def __init__(self):
         super().__init__()
         pass

--- a/rio/commands/nullauto.py
+++ b/rio/commands/nullauto.py
@@ -2,6 +2,7 @@ import commands2
 from subsystems import drivetrain
 
 
+class Auto(commands2.SequentialoCommandGroup):
     def __init__(self):
         super().__init__()
         pass

--- a/rio/commands/nullauto.py
+++ b/rio/commands/nullauto.py
@@ -1,8 +1,29 @@
 import commands2
-from subsystems import drivetrain
+
+from commands.flybywire import FlyByWire
 
 
-class Auto(commands2.SequentialoCommandGroup):
-    def __init__(self):
+class NullAuto(commands2.SequentialCommandGroup):
+    def __init__(self, drivetrain) -> None:
+        """
+        Fill this out.
+        """
         super().__init__()
-        pass
+
+        # list of commands.
+        self.addCommands(
+            FlyByWire(
+                drivetrain,
+                lambda: self.getNum(0),
+                lambda: self.getNum(1),
+                lambda: self.getNum(0),
+            ),
+        )
+
+    def getNum(self, num):
+        """
+        Replace me later.
+        Typing weirdness.
+        """
+
+        return num

--- a/rio/commands/nullauto.py
+++ b/rio/commands/nullauto.py
@@ -1,7 +1,6 @@
 import commands2
 from subsystems import drivetrain
 
-class Auto(commands2.SequentialoCommandGroup):
 
     def __init__(self):
         super().__init__()

--- a/rio/robotcontainer.py
+++ b/rio/robotcontainer.py
@@ -55,5 +55,5 @@ class RobotContainer:
         pass
 
     def getAutonomousCommand(self) -> commands2.Command:
+        self.chooser.setDefaultOption("Auto", self.Auto)
         return self.chooser.getSelected()
-        self.chooser.setDefaultOption("Auto", self.Auto) 

--- a/rio/robotcontainer.py
+++ b/rio/robotcontainer.py
@@ -13,7 +13,7 @@ from subsystems.drivetrain import Drivetrain
 from constants.constants import getConstants
 
 # Auto
-from commands.nullauto import Auto
+from commands.nullauto import NullAuto
 
 
 class RobotContainer:
@@ -34,7 +34,11 @@ class RobotContainer:
         # The robot's subsystems
         self.drivetrain = Drivetrain()
 
+        # Configure button bindings
         self.configureButtonBindings()
+
+        # Setup all autonomous routines
+        self.configureAutonomous()
 
         # set up default drive command
         self.drivetrain.setDefaultCommand(
@@ -54,6 +58,15 @@ class RobotContainer:
         """
         pass
 
+    def configureAutonomous(self):
+        # Create a sendable chooser
+        self.autoChooser = wpilib.SendableChooser()
+
+        # Add options for chooser
+        self.autoChooser.setDefaultOption("Null Auto", NullAuto(self.drivetrain))
+
+        # Put the chooser on the dashboard
+        wpilib.SmartDashboard.putData("Autonomous", self.autoChooser)
+
     def getAutonomousCommand(self) -> commands2.Command:
-        self.chooser.setDefaultOption("Auto", self.Auto)
-        return self.chooser.getSelected()
+        return self.autoChooser.getSelected()

--- a/rio/robotcontainer.py
+++ b/rio/robotcontainer.py
@@ -56,4 +56,4 @@ class RobotContainer:
 
     def getAutonomousCommand(self) -> commands2.Command:
         return self.chooser.getSelected()
-        self.chooser.setDefaultOption("Auto", self.Auto) 
+        self.chooser.setDefaultOption("Auto", self.Auto)

--- a/rio/robotcontainer.py
+++ b/rio/robotcontainer.py
@@ -12,6 +12,9 @@ from subsystems.drivetrain import Drivetrain
 # Constants
 from constants.constants import getConstants
 
+# Auto
+from commands.nullauto import Auto
+
 
 class RobotContainer:
     """
@@ -53,3 +56,4 @@ class RobotContainer:
 
     def getAutonomousCommand(self) -> commands2.Command:
         return self.chooser.getSelected()
+        self.chooser.setDefaultOption("Auto", self.Auto) 

